public/app/core/components/AppChrome/AppChrome.tsx

```diff
@@ -136,14 +151,15 @@ export function AppChrome({ children }: Props) {
   );
 }
 
-const getStyles = (theme: GrafanaTheme2, hasActions: boolean) => {
+const getStyles = (theme: GrafanaTheme2, hasActions: boolean, isExtensionSidebarOpen: boolean) => {
+  // styles depend if the extension sidebar is open or not. if it is closed, we want the body to scroll
   return {
     content: css({
       display: 'flex',
       flexDirection: 'column',
       paddingTop: hasActions ? TOP_BAR_LEVEL_HEIGHT * 2 : TOP_BAR_LEVEL_HEIGHT,
       flexGrow: 1,
-      height: 'auto',
+      ...(isExtensionSidebarOpen ? { height: '100vh', overflow: 'hidden' } : { height: 'auto' }),
```

Comment:
any chance you can do this with an override class instead (like the most other CSS logic here we try to use different classes for different states instead of building the classes dynamically)

Reply:
Good idea, done in 962d7f3103532d66b8af9e317e96541973415f56